### PR TITLE
Added instructions in documentation for how to compile and build personal designs

### PR DESCRIPTION
**Pull request by WhiteNinjaZ**
@acomodi This pull request adds instructions for how to create personal design files and build them through the Symbiflow toolchain. It also gives a detailed walkthrough of how the Makefiles throughout Symbiflow work and how to create ones own.

### CODE REVIEW

File: README.rst
Comment (acomodi) [inline README.rst:27]: Long lines should be divided into multiple smaller lines

File: docs/Understanding-Makefile.rst
Comment (acomodi) [inline docs/Understanding-Makefile.rst:1]: I think that, rather than the Makefiles, which is just a way of automatically executing some commands (e.g. a bash script or python could be used instead), what would be important to describe are the various commands that are being run, e.g.:
- what does `symbiflow_synth` internally do, etc.?
- what are the arguments that are to be provided, and what do they mean?

I'd say that the important thing is to not focus on how to use Makefiles, but rather on what are the required commands to run the toolchain to completion. Possibly this

Comment (acomodi) [reply docs/Understanding-Makefile.rst:1]: On a side note, it would be good to use the naming of the file consistent with other doc files, and have something like `understanding-makefiles`. Note that, per the previous comment, the name of this file will need to change.

File: docs/Understanding-Makefile.rst
Comment (acomodi) [inline docs/Understanding-Makefile.rst:85]: This is not entirely true as SystemVerilog is not fully supported in Yosys

File: docs/personal-designs.rst
Comment (acomodi) [inline docs/personal-designs.rst:61]: As above, SystemVerilog is not fully supported.

The statement about the `top` module is not entirly correct, as for instance, `symbiflow_synth` has an argument to define [the top-level module name](https://github.com/SymbiFlow/symbiflow-arch-defs/blob/39de38b90ed61fd3fd6e162f8296beeef322759f/xc/xc7/toolchain_wrappers/symbiflow_synth#L31-L38)

File: docs/personal-designs.rst
Comment (acomodi) [inline docs/personal-designs.rst:67]: A description of what XDC, PCF and SDC file should end up in the former `understanding-makefiles` document.

File: docs/personal-designs.rst
Comment (acomodi) [inline docs/personal-designs.rst:1]: I'd just rename this `Building Custom Designs`

**Review by acomodi — CHANGES_REQUESTED** [review]
Hi @WhiteNinjaZ! Thanks for your PR. I have added some comments that would need to be solved and, once they have, I'll get another review pass.

To ease adding comments and also having the docs more maintainable, all the long lines should be split in multiple lines. I think that a good measure is to keep lines below the 100 characters threshold.

File: docs/personal-designs.rst
Comment (WhiteNinjaZ) [reply docs/personal-designs.rst]: I believe I have that already in customizing-makefiles->Constraint files. Are you saying I should explain how the XDC, PCF and SDC files should be written or something else? Could you clarify?

Comment (acomodi) [reply docs/personal-designs.rst]: Yeah, I meant to add some notes on what are those constraint files and what content they should have. For instance, the XDC defines can define both IOPAD constraints, IOSETTINGS and also clock constraints.

A PCF file can define the constraints for IOPADs only, while SDC is used for clock constraints needed for VTR.

If you, for instance, set only an XDC constraint file, containing both IOPAD constraints and clock constraints, an SDC will be automatically generated to provide clock constraints to VTR. Otherwise, you need to provide the SDC explicitly.

So there are two main ways of adding constraints:
1. PCF (iopad) +SDC (clock)
2. XDC (both iopads and clock constraints)

File: docs/Understanding-Makefile.rst
Comment (WhiteNinjaZ) [reply docs/Understanding-Makefile.rst]: Thanks for the suggestion. I think that documentation for how the tool chain commands work is very import, but I also think that instructions on how the Makefiles in symbiflow work and how to modify them for personal designs is important. I have just finished creating two separate sections within the documentation describing how to implement both of these methods and have pushed them to this PR. I believe that separate sections on both topics are beneficial for three reasons: 1) Users who are just starting with symbiflow/linux may be more comfortable with modifying existing makefiles then they would be to create there own scripts 2) The makefile documentation provides valuable insights that might prove useful for debugging issues encountered while building the provided examples 3) by separating instructions into two parts, users are empowered with the option to chose which method (ether makefile, python, or bash script) works best for them. Let me know what you think.

File: docs/personal-designs.rst
Comment (acomodi) [inline docs/personal-designs.rst:1]: I think this file might not be strictly needed as it is quite similar on how to build and run pre-existing examples.

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:22]: nitpick - remove extra new-lines

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:36]: I think it is better to avoid specifying an unsupported and untested HDL
```suggestion
    | -v   | A list of paths to verilog files for the design |
```

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:47]:
```suggestion
verilog HDL files is below. The design is built for a basys3 board which comes from the artix7 
```

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:56]: The content in the parenthesis is covered in the below note and is superfluous in here

```suggestion
an .eblif file, a few verilog netlists that describe the gate level design for your project, and a log
```

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:68]:
```suggestion
Place and Route
```

File: docs/understanding-commands.rst
Comment (acomodi) [inline docs/understanding-commands.rst:115]:
```suggestion
    | -d   | Fabric specification (xc7a50t_test)                 |
```

**Review by acomodi — COMMENTED** [review]
I have added some more detailed comments. I believe this is getting to a mergeable state. Once all the comments are addressed, please re-request another review for a possible final pass

File: docs/customizing-makefiles.rst
Comment (WhiteNinjaZ) [reply docs/customizing-makefiles.rst:300]: I tend to agree. The only reason I included the XDC+PCF+SDC was because I saw that for some reason the [Linux litex demo](https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/linux_litex_demo/Makefile) appears to use this combination. I will remove this though to avoid confusion.

File: docs/personal-designs.rst
Comment (WhiteNinjaZ) [reply docs/personal-designs.rst:1]: Fair point. This file was a little repetitive. I went ahead and cut out a few parts from it and referenced other pieces of the documentation instead of re-stating what was already said.
I kept the Building Custom designs file because I think that having a separate section for personal designs will help make it very clear to users that they can run there own designs through the toolchain (something I was not entirely aware I could do when I first read the documentation). I also think there are some pieces of useful information within this file. That being said I don't think it would be to hard to move that information into already existing sections. Let me know what you think.

File: docs/customizing-makefiles.rst
Comment (WhiteNinjaZ) [reply docs/customizing-makefiles.rst]: Not that I am aware of. I have been looking through several tutorials and it might be possible to link to specific lines of code using a [parsed-literal block](https://stackoverflow.com/questions/44527391/how-to-include-an-internal-reference-in-a-code-block) instead of a code-block. However, that would come with several drawbacks including the loss of line numbers, the inability to highlight lines of interest within the code, and the loss of bash font. I did the next best thing and linked each reference to a line of code to the code block itself. I think this worked out quite well.

**Comment by WhiteNinjaZ** [comment]
@acomodi Please see my comments above. Other, than the three things I mentioned I believe I have dealt with all of the issues you brought to my attention.

File: docs/customizing-makefiles.rst
Comment (acomodi) [reply docs/customizing-makefiles.rst]: Ok, I guess that it makes sense to leave this as is than

**Review by acomodi — APPROVED** [review]
LGTM, I have resetarted CI once again as there was a failure in one of the builds due to some infrastructure issue I believe. Once green I'll merge

**Pull request merged**
